src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/resettlementassessmentstrategies/AbstractResettlementAssessmentStrategy.kt

```diff
@@ -46,15 +48,7 @@ abstract class AbstractResettlementAssessmentStrategy<T, Q>(
   IResettlementAssessmentStrategy where T : Enum<*>, T : IAssessmentPage, Q : Enum<*>, Q : IResettlementAssessmentQuestion {
 
   override fun appliesTo(pathway: Pathway): Boolean {
-    return if (useYaml) {
-      if (pathway == Pathway.ACCOMMODATION) {
-        false
-      } else {
-        pathway == this.pathway
-      }
-    } else {
-      pathway == this.pathway
-    }
+    return (useYaml && pathway !in yamlPathways) && this.pathway == pathway
```

Comment:
```
fun isApplicableBasedOnYaml(pathway: Pathway): Boolean {
    return useYaml && pathway!in yamlPathways
}

fun isPathwayMatch(pathway): Boolean {
    return this.pathway == pathway
}

override fun appliesTo(pathway: Pathway): Boolean {
    return isApplicableBasedOnYaml(pathway) && isPathwayMatch(pathway)
}
```

might be easier to digest?

Reply:
I didn't see this comment until I refactored it in I slightly different way 😅 Agreed that it was hard to read though